auth_session_timeout/models/ir_config_parameter.py

```diff
@@ -13,24 +12,18 @@
 class IrConfigParameter(models.Model):
     _inherit = 'ir.config_parameter'
 
-    @tools.ormcache(skiparg=0)
-    def get_session_parameters(self, db):
-        param_model = self.pool['ir.config_parameter']
-        cr = self.pool.cursor()
-        delay = False
-        urls = []
-        try:
-            delay = int(param_model.get_param(
-                cr, SUPERUSER_ID, DELAY_KEY, 7200))
-            urls = param_model.get_param(
-                cr, SUPERUSER_ID, IGNORED_PATH_KEY, '').split(',')
-        finally:
-            cr.close()
-        return delay, urls
+    @api.model
+    @tools.ormcache('self.env.db')
```

Comment:
@lasley, about https://github.com/OCA/server-tools/pull/580#discussion_r94074703 (I answer here so this does not get hidden)...

I learned the hard way to use ormcache when developing [`website_sale_wishlist`](https://github.com/OCA/e-commerce/tree/9.0/website_sale_wishlist). It turns out it's a pretty neat tool. It creates a key by the arguments you pass to it, and stores the method result in a cache attached to your current `self.env`, so further calls to the method return the cached result instead of executing the method again and again.

It's most useful when a method is CPU-expensive, is called quite often, and it returns a predictable result based on the incoming arguments/environment.

In old api, we had the `skipargs`, which was used to skip a number of arguments and use the rest to make a cache. For instance, if your method had `cr, uid, ids, key, context=None` and the result was the same no matter the `cr, uid`, you'd say `skipargs=2`.

Since v9, [we have the new notation where you give the name of the arguments that form the cache key](https://github.com/odoo/odoo/blob/9.0/openerp/tools/cache.py#L36). Since `self` is always the first argument, [you can use that one (and properties of it) to form a key](https://github.com/OCA/e-commerce/blob/9.0/website_sale_wishlist/models/product_template.py#L10-L17).

When you change your `env` (like `self.with_context(goal="I wanna be a pirate")`), cache is empty. This is logical because a different context/cr/uid usually yields different results almost everywhere. However, when you apply cache to a method, [it gets a `clear_cache()` method](https://github.com/odoo/odoo/blob/9.0/openerp/tools/cache.py#L36d) that [you can use later to clear it without having to change your environment](https://github.com/OCA/e-commerce/blob/9.0/website_sale_wishlist/models/product_wishlist.py#L93-L100).